src/app/Library/CrudPanel/Traits/HasViewNamespaces.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\CrudPanel\Traits;
+
+trait HasViewNamespaces
+{
+    private $viewNamespaces = [];
+
+    /**
+     * Return the namespaces stored for the given domain.
+     *
+     * @param  string  $domain  (eg. fields, filters, buttons)
+     * @return array
+     */
+    private function getViewNamespacesFor(string $domain)
+    {
+        return $this->viewNamespaces[$domain] ?? [];
+    }
+
+    /**
+     * Return all the view namespaces including the ones stored in the laravel config files.
+     *
+     * @param  string  $domain  (eg. fields, filters, buttons)
+     * @return array
+     */
+    public function getAllViewNamespacesFor(string $domain)
+    {
+        $viewNamespacesFromConfig = $this->getViewNamespacesFromConfigFor($domain);
+
+        return array_unique(array_merge($viewNamespacesFromConfig, $this->getViewNamespacesFor($domain)));
+    }
```

Comment:
I really don't like the "all" in the name. Since it's already in plural, it's kind of obvious it's going to get _all_ of them. No chance it's going to get _some_ of them, otherwise the name would say which ones.

```suggestion
    public function getViewNamespacesFor(string $domain)
    {
        $viewNamespacesFromConfig = $this->getViewNamespacesFromConfigFor($domain);

        return array_unique(array_merge($viewNamespacesFromConfig, $this->viewNamespaces[$domain] ?? []);
    }
```